repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,174 @@
+# Deploy Repairnator in Kubernetes
+Note: to store data you can either deploy activeMQ on cloud described at the bottom of this README or if you have a mongodb outside the cluster you can map it as an external service named as mongo (since all of the default mongo k8s service name is named "mongo" if you don't like a lot changes). More details for how this is done is described in [this link](ttps://cloud.google.com/blog/products/gcp/kubernetes-best-practices-mapping-external-services)
+
+![Design](K8sRepairnatorDesign.jpg)
+
+## Prerequisites
+
+* kubectl
+* a working k8s cluster (preferable high cpu capacity)
+* gcloud (for creating the cluster mentoned above and creating mongodb if you don't have them already)
+
+## Setup ActiveMQ
+
+To deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+```
+	kubectl create -f /queue-for-buildids/activemq.yaml
+```
+To access to the web interface
+```
+	kubectl get pods 
+```
+It should look like activemq-XXXXXXX-XXXXX. Then 
+```
+	kubectl exec -it activemq-XXXXXXX-XXXXX bash
+```
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+```
+	kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+```
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+```
+	python publisher.py -d queue-name message-1 message-2 message3 ... message-N
+```
+
+queue-name is according to the format /queue/name. For instance /queue/scanner will send message to the scanner queue.
+You can use this later to manually input build id to the pipeline queue or project to the scanner queue
+
+## Setup scanners and pipelines
+
+To deploy the scanner first go into folder "scanner-dockerimage" then directly build and push the image to docker hub(enviroment variables can be specified or changed later in the "repairnator-scanner.yaml" file. With docker 
+```
+	docker build -t repairnator-scanner:tagname .
```

Comment:
in which folder must we be before starting this command?

add `cd foo/bar`?

Reply:
Added one line about that in README. The start folder is "repairnator/kubernetes-support"